components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -290,6 +304,58 @@ func (w *NotebookWebhook) InjectDecoder(d *admission.Decoder) error {
 	return nil
 }
 
+// maybeRestartRunningNotebook evaluates whether the updates being made cause notebook pod to restart.
+// If the restart is caused by updates made by the mutating webhook itself to already existing notebook,
+// and the notebook is not stopped, then these updates will be blocked until the notebook is stopped.
+// Returns the mutated notebook, a flag whether there's a pending restart (to apply blocked updates), and an error value.
+func (w *NotebookWebhook) maybeRestartRunningNotebook(req admission.Request, mutatedNotebook *nbv1.Notebook) (*nbv1.Notebook, bool, error) {
```

Comment:
nitpick of all nitpicks: all the other functions in this file start with capital character :upside_down_face: 
```suggestion
func (w *NotebookWebhook) MaybeRestartRunningNotebook(req admission.Request, mutatedNotebook *nbv1.Notebook) (*nbv1.Notebook, bool, error) {
```

Reply:
lowercase first letter means function is `private`; I don't need it public, so lowercase it it

Reply:
> being able to access unexported (read: lowercase, sigh) symbols

https://fasterthanli.me/articles/i-want-off-mr-golangs-wild-ride

Reply:
Hehe, right, shame on me :slightly_smiling_face: